### PR TITLE
added scala 3 to cross versions

### PR DESCRIPTION
**Pull request by sviezypan**
- added scala 3 version to build.sbt
- upgrade sttp to sttp3
- changed in tests because of breaking changes in sttp3
- upgrade sbt version

### CODE REVIEW

**Comment by sviezypan** [comment]
@jeremyrsmith Hello, I am trying to add scala 3 support to uzhttp because I need scala3 version of this library for zio-zmx. I was hoping you could help me by reviewing this PR? - I assume you're the maintainer.

I am not really much experienced in the build stuff. I verified this work just by "sbt publishLocal" and by using new version in zio-zmx.

File: src/main/scala-3/uzhttp/header/CompatMap.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package uzhttp.header
+
+trait CompatMap[K, V] extends scala.collection.immutable.Map[K, V]
```
Comment (jeremyrsmith) [inline src/main/scala-3/uzhttp/header/CompatMap.scala:3]: Is this used anywhere?

Comment (sviezypan) [reply src/main/scala-3/uzhttp/header/CompatMap.scala:3]: [Headers](https://github.com/polynote/uzhttp/blob/master/src/main/scala/uzhttp/header/Headers.scala) class extends CompatMap

Comment (jeremyrsmith) [reply src/main/scala-3/uzhttp/header/CompatMap.scala:3]: Oh, yes of course 🤦

**Comment by jeremyrsmith** [comment]
LGTM. Thanks @sviezypan!

**Comment by sviezypan** [comment]
thanks @jeremyrsmith could we release this new version?

**Comment by jeremyrsmith** [comment]
@sviezypan It's on its way to Maven Central now! Thanks!

**Pull request merged**